pennylane_lightning/core/src/simulators/lightning_kokkos/gates/BasicGeneratorFunctors.hpp

```diff
@@ -196,17 +195,17 @@ void applyGenSingleExcitation(
     const std::size_t num_qubits, const std::vector<std::size_t> &wires,
     [[maybe_unused]] const bool inverse = false,
     [[maybe_unused]] const std::vector<PrecisionT> &params = {}) {
-    applyNC2Functor(
-        ExecutionSpace{}, arr_, num_qubits, wires,
-        KOKKOS_LAMBDA(Kokkos::View<Kokkos::complex<PrecisionT> *> arr,
-                      const std::size_t i00, const std::size_t i01,
-                      const std::size_t i10, const std::size_t i11) {
-            arr(i00) = 0.0;
-            arr(i01) *= Kokkos::complex<PrecisionT>{0.0, 1.0};
-            arr(i10) *= Kokkos::complex<PrecisionT>{0.0, -1.0};
-            arr(i11) = 0.0;
-            kokkos_swap(arr(i10), arr(i01));
-        });
+    auto core_function = KOKKOS_LAMBDA(
+        Kokkos::View<Kokkos::complex<PrecisionT> *> arr, const std::size_t i00,
+        const std::size_t i01, const std::size_t i10, const std::size_t i11) {
+        arr(i00) = 0.0;
+        arr(i01) *= Kokkos::complex<PrecisionT>{0.0, 1.0};
+        arr(i10) *= Kokkos::complex<PrecisionT>{0.0, -1.0};
+        arr(i11) = 0.0;
```

Comment:
Is  `arr(i11)` a floating data or complex one?

Reply:
it is a [Kokkos complex](https://kokkos.org/kokkos-core-wiki/API/core/utilities/complex.html) which can be constructed with just a real float